requirements.txt

```diff
@@ -1,71 +1,2 @@
-chardet==3.0.4            # via requests
-click==7.1.2              # via black, pip-tools
-cycler==0.10.0            # via matplotlib
-distlib==0.3.1            # via virtualenv
-filelock==3.0.12          # via virtualenv
-flake8==3.8.3             # via -r requirements.in
-future==0.18.2            # via torch
-graphviz==0.14.1          # via -r requirements.in
-identify==1.4.29          # via pre-commit
-idna==2.10                # via requests
-importlib-metadata==1.7.0  # via flake8, pre-commit, virtualenv
-importlib-resources==3.0.0  # via pre-commit, virtualenv
-isort[requirements]==5.4.2  # via -r requirements.in
-joblib==0.16.0            # via scikit-learn
-kiwisolver==1.2.0         # via matplotlib
-matplotlib==3.3.2         # via netcal
-mccabe==0.6.1             # via flake8
-mypy-extensions==0.4.3    # via black
-mysqlclient==2.0.1        # via -r requirements.in
-netcal==1.1.2             # via -r requirements.in
-nodeenv==1.5.0            # via pre-commit
-nose==1.3.7               # via -r requirements.in
-numpy==1.19.1             # via -r requirements.in, matplotlib, netcal, pandas, patsy, scikit-learn, scipy, statsmodels, torch
-pandas==1.1.1             # via -r requirements.in, statsmodels
-pathspec==0.8.0           # via black
-patsy==0.5.1              # via statsmodels
-pillow==7.2.0             # via matplotlib
-pip-tools==5.3.1          # via -r requirements.in
-plotly==4.9.0             # via -r requirements.in
-pre-commit==2.7.1         # via -r requirements.in
-py4j==0.10.9              # via pyspark
-pycodestyle==2.6.0        # via flake8
-pydot==1.4.1              # via -r requirements.in
-pyflakes==2.2.0           # via flake8
-pyparsing==2.4.7          # via matplotlib, pydot
-pyspark-stubs==3.0.0      # via -r requirements.in
-pyspark==3.0.0            # via -r requirements.in, pyspark-stubs
-python-dateutil==2.8.1    # via matplotlib, pandas
-pytz==2020.1              # via pandas
-pyyaml==5.3.1             # via pre-commit
-regex==2020.7.14          # via black
-requests==2.24.0          # via -r requirements.in
-retrying==1.3.3           # via plotly
-scikit-learn==0.23.2      # via -r requirements.in, netcal
-scipy==1.5.2              # via netcal, scikit-learn, statsmodels
-six==1.15.0               # via patsy, pip-tools, plotly, python-dateutil, retrying, virtualenv
-sqlalchemy==1.3.19        # via -r requirements.in
-statsmodels==0.12.0       # via -r requirements.in
-threadpoolctl==2.1.0      # via scikit-learn
-toml==0.10.1              # via black, pre-commit
-torch==1.6.0              # via netcal
-tqdm==4.49.0              # via netcal
-typed-ast==1.4.1          # via black
-typing-extensions==3.7.4.3  # via black
-urllib3==1.25.10          # via requests
-virtualenv==20.0.31       # via pre-commit
-zipp==3.1.0               # via importlib-metadata, importlib-resources
- 
-# The following packages are considered to be unsafe in a requirements file:
-# pip
+Flask==1.1.1
+mysql-connector==2.2.9
```

Comment:
Just taking a stab here, but is it really necessary to have the mysql-connector here? It looks as if you are using the MariaDb image but not sure if this is required to run everything.

Reply:
This is part of the test.py to show a simple creation of a docker image. It is not needed for the assignment specifically. I should consider removing them both.